lib/cylc/graphing.py

```diff
@@ -259,11 +261,44 @@ def add_subgraph(self, nbunch=None, name=None, **attr):
 
         return subgraph
 
+    def set_def_style(self, fgcolor, bgcolor, def_node_attr=None):
+        """Set default graph styles.
+
+        Depending on light/dark desktop color theme.
+        """
+
+        if def_node_attr is None:
+            def_node_attr = {}
+
+        # Transparent graph bg - let the desktop theme bg shine through.
+        self.graph_attr['bgcolor'] = '#ffffff00'
+
+        fg, bg = str(fgcolor), str(bgcolor)
+        # 3-digit hex color codes are not recognized.
+        if len(fg) == 4:
+            fg = '#' + fg[1:2]*2 + fg[2:3]*2 + fg[3:4]*2
+        if len(bg) == 4:
+            bg = '#' + bg[1:2]*2 + bg[2:3]*2 + bg[3:4]*2
+
```

Comment:
@oliver-sanders - above is my 3-char hex color code fix.  I'll probably switch to your prettier version tomorrow!

Reply:
Ah, diddn't spot that!

Reply:
(OK, I haven't used this everywhere it's needed - specifically in the GUI code)